zowe/zowe-cli
# Remove the 'zowe profiles' command command handlers & APIs

**What It Does**


This PR removes the command handlers for the `zowe profiles` command group.

Also:
- Numerous APIs, interfaces, and constants that supported V1 profiles were also removed.

- Three classes (AbstractProfileManager which was extended by BasicProfileManager, which was extended by CliProfileManager) were consolidated into the one CliProfileManager class.

- The `--show-inputs-only` options and the `config report-env` commands no longer report V1 profile information.

- Replaced the term "Team configuration" with "Zowe client configuration" in the `zowe config report-env` command.

- Changed a reference in the `config convert-profiles` command  from "V2 profiles" to "TeamConfig".

- Removed the display of the NVM version number in the `config report-env` command. Newer versions of NVM display a GUI popup message when launched as a subprocess, which complains that NVM should be run from a terminal (version 1.1.12 displays this behavior). The popup blocks further output until the user responds to the popup. NVM is *NOT* used by customers frequently enough to warrant an exhaustive effort to keep that piece of information in the report. For awareness, the NVM dialog is shown below.

    ![image](https://github.com/zowe/zowe-cli/assets/37119146/1b6f9231-600d-4813-8263-34f98137c3be)

**How to Test**



1. Commands should continue to run as before when no configuration exists or a team configuration exists. Commands should ignore V1 profiles, even when they are the only available configuration. In particular, if only V1 profiles exist, running a Zowe command now will prompt the user for connection information.
    ```
    > zowe files list data-set "SYS1.PARMLIB*"
    Enter the host name of your service: YourHostName
    Enter the user name for your service (will be hidden):
    Enter the password for your service (will be hidden):
    ```
2. Use `--show-inputs-only`, `config report-env`, & `config convert-profiles` and observe that no references to V1 or V2 profiles are displayed.

#### Follow-ups



- The "ProfileIO" class must be renamed to "V1ProfileConversion", and all write functions must be removed.
- Any additional V1 logic that is no longer called should be removed.

**Review Checklist**
I certify that I have:


**Additional Comments**
